pkg/signatures/public_key_fetcher.go

```diff
@@ -53,6 +53,12 @@ func init() {
 // NOTE: No error will be returned when the image has no signature available. All occurring errors will be logged.
 func (c *cosignPublicKeySignatureFetcher) FetchSignatures(ctx context.Context, image *storage.Image,
 	registry registryTypes.Registry) ([]*storage.Signature, error) {
+	// Short-circuit for images that do not have V2 metadata associated with them. These would be older images manifest
+	// schemes that are not supported by cosign, like the docker v1 manifest.
+	if image.GetMetadata().GetV2() == nil {
```

Comment:
@dhaus67 Something just hit me, this will make it so that metadata must be pulled before signatures. This is desired right?

Reply:
Yeah, I've checked the two places where the signature fetching is done ,would be a) [the enrichment for local scanner](https://github.com/stackrox/stackrox/blob/master/sensor/common/scan/scan.go#L39) and b) [the image enricher](https://github.com/stackrox/stackrox/blob/master/pkg/images/enricher/enricher_impl.go#L171). Both times we do just fetch the image metadata beforehand (we short-circuit in both places when image metadata cannot be fetched).